invenio_records_resources/services/files/schema.py

```diff
@@ -85,7 +85,18 @@ def fields_from_file_obj(self, data, **kwargs):
         if data.file:
             # mandatory fields
             data["storage_class"] = data.file.storage_class
-            data["uri"] = data.file.uri  # TODO: only serialize for external?
+            data["uri"] = data.file.uri
+
+            # If Local -> remove uri as it contains internal file storage info
+            if data["storage_class"] == TransferType.LOCAL:
```

Comment:
Can we add a method in the Transfer class, for example "local_uri" (I dont like this name). To avoid coupling the TransferTypes (data layer) to the service serialization?

Reply:
Maybe @TLGINO understood, but I am not sure I did fully :) It sounds like a good idea, could you elaborate what you mean? :) Thank you!

Reply:
Sorry, I should have explained better. In invenio-records-resources the [TransferType] class has a method to specify if the transfer has finished or not. I was wondering if we could make a similar method to enable/disable serialization (e.g. is_serializable). That way we would abstract this behavior (i.e. in case we create more transfertype we do not need to modify the serialization/service schema). It would also simplify the code:

```python
if not TransferType(data["storage_class"]).is_serializable:
                 data.pop("uri")
```